hazelcast/src/main/java/com/hazelcast/map/impl/querycache/accumulator/DefaultCyclicBuffer.java

```diff
@@ -33,37 +33,38 @@
  */
 public class DefaultCyclicBuffer<E extends Sequenced> implements CyclicBuffer<E> {
 
-    private static final long UNAVAILABLE = -1L;
+    private static final long UNSET = -1L;
 
     private int capacity;
     private E[] buffer;
     private AtomicLong headSequence;
```

Comment:
This smells like a design problem. The class is lock free and now get some locks.

Reply:
put a new assertion to have much info about failure and removed locks